fiware_cloto/environments/tests/test_environment_controller.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+# -*- encoding: utf-8 -*-
+#
+# Copyright 2015 Telefónica Investigación y Desarrollo, S.A.U
+#
+# This file is part of FI-WARE project.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at:
+#
+
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For those usages not covered by the Apache version 2.0 License please
+# contact with opensource@tid.es
+#
+__author__ = 'gjp'
+from django.test import TestCase
+
+from fiware_cloto.environments import environment_controller
+
+
+class ClientTests(TestCase):
+
+    def setUp(self):
+        """sets up the environment to work with."""
+
+
+    def test_clean_no_environments(self):
+        """test_clean_environments should check that there are no environments to clean."""
+        try:
+            error = environment_controller.clean_environments()
```

Comment:
If I understand the tests, you are making a real e2e test here, due to you are calling the Popen process to search and kill the process, there is no other alternative to mock the Popen process?

Anyway, you are checking that there is no process but if there is a process?